Assets/Scripts/Gun.cs

```diff
@@ -21,12 +21,15 @@ void Awake()
 
 	void Update ()
 	{
-		// If the fire button is pressed...
-		if(Input.GetButtonDown("Fire1"))
+        if (Pauser.IsPaused)
+            return;
+        // If the fire button is pressed...
+        if (Input.GetButtonDown("Fire1"))
 		{
 			// ... set the animator Shoot trigger parameter and play the audioclip.
 			anim.SetTrigger("Shoot");
-			audio.Play();
+            GetComponent<AudioSource>().volume = SettingsService.GetVolumeEffect();
```

Comment:
This is probably because of the update to the new Unity, but, how do you think this could be made more efficient?

Reply:
I think it would be better to store the "GetComponent <AudioSource> ()" in the Awake part. Which avoids calling the GetComponent <T> () every time. I correct that too.

Reply:
The same error was made in MusicPlayer.cs, so i correct that. Thanks.